### PR TITLE
Add test coverage

### PR DESCRIPTION
**Pull request by peifeng-unity**
## Proposed change(s)

Add test coverage check in the Yamato job. The current criteria are 0% and we should gradually increase the criteria in each of the future PR until it reaches the standard. For example, 3% for each PR.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

Provide any relevant links here.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Documentation update
- [x] Other: update Yamato to check the test coverage.

## Testing and Verification

Tests are shown below in the Yamato pipeline in this PR.

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [ ] Followed the style guidelines as described in the [Contribution Guidelines](../CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

File: .yamato/yamato-config.yml
Comment (mrpropellers) [inline .yamato/yamato-config.yml:27]: We should keep one or the other of these two blocks or we'll end up running redundant tests. There were two reasons I used the original `push.branch eq "dev"` logic instead of `pull_request.target eq "dev"`
1. `pull_request.target` will trigger a yamato run not just when a PR is submitted, but on every subsequent update to that PR too, which can cause a lot of churn occasionally. But, maybe the threat of churn is a good forcing function to encourage devs to run these tests locally before submitting their PR...
2. Yamato has a tendency to become unstable and get backed up, which can make testing times go from ~5 minutes to 45+ minutes. If we block merges pending Yamato results, we'll definitely run into times where a PR is approved, but unable to be merged for an hour because of Yamato issues... We could mitigate this, though, by allowing manual merge overrides in  cases where Yamato is acting extra slow and we want to get something in faster.
This is totally up for discussion, I don't have a strong opinion either way. I set it up the way I did originally to reduce friction, but there's definitely a lot of benefits to forcing PRs to wait for test results.

Comment (peifeng-unity) [reply .yamato/yamato-config.yml:27]: Ah, thanks, Devin! I didn't know all these potential issues on Yamato. It makes sense to me that we shouldn't be blocked by the Yamato issue. I personally often forget to rerun the unit tests after dealing with PR comments, so I thought it might be a concern that the unit tests will not pass if we don't force the checking and it will be a huge trouble for the next committer.

Do you think it would be better if we can manually trigger the Yamato jobs right before merging them into the dev branch? (I am not sure if Yamato can do this or not though)

Comment (mrpropellers) [reply .yamato/yamato-config.yml:27]: > Do you think it would be better if we can manually trigger the Yamato jobs right before merging them into the dev branch? (I am not sure if Yamato can do this or not though)  
  
Yamato can do this, but I think we should think of anything that requires manual action to be not reliable enough for consideration.  In the case where we want tests to run _before_ merge, `cancel_old_ci: true` guarantees that we won't have to wait on stale tests. If a PR revision is submitted before the old tests are finished, the currently running tests are discarded and a new CI run is initiated with the fresh revision.

Comment (peifeng-unity) [reply .yamato/yamato-config.yml:27]: Okay, get you! that makes sense.

**Review by mrpropellers — APPROVED** [review]

**Pull request merged**
